client/dashboard/sites/overview/index.tsx

```diff
@@ -158,12 +208,24 @@ function SiteOverview( {
 					spacing={ spacing }
 					alignment="flex-start"
 				>
-					<LatestActivityCard site={ site } isCompact={ isSmallViewport } />
-					{ ! isSelfHostedJetpackConnectedSite && ! site.is_wpcom_staging_site && (
```

Comment:
Note that the `! isSelfHostedJetpackConnectedSite` condition was added in https://github.com/Automattic/wp-calypso/pull/105832 to prevent Domain card CTA landing in dead links.

That said, self-hosted sites should be eligible to the domain transfer upsell card https://github.com/Automattic/wp-calypso/pull/104899.

Since this PR removes the Domain card, we can lift the restriction on self-hosted sites on rendering this section.